pkg/codegen/nodejs/gen.go

```diff
@@ -121,7 +121,7 @@ func (mod *modContext) tokenToModName(tok string) string {
 
 func (mod *modContext) namingContext(pkg *schema.Package) (namingCtx *modContext, pkgName string, external bool) {
 	namingCtx = mod
-	if pkg != nil && pkg != mod.pkg {
+	if pkg != nil && pkg.Name != mod.pkg.Name {
```

Comment:
💥  Really need to understand this one. It might not be correct. But codegen breaks badly without the change.